include/quicly/sentmap.h

```diff
@@ -303,6 +301,15 @@ inline void quicly_sentmap_init_iter(quicly_sentmap_t *map, quicly_sentmap_iter_
         iter->p = (quicly_sent_t *)&quicly_sentmap__end_iter;
         iter->count = 0;
     }
+
+    /* Retire entries older than 4 PTO, unless the connection is alive and the number of packets in the sentmap is below 32 packets.
```

Comment:
This is a quick and dirty hack to prevent old packets from getting deemed as lost under a situation in which consecutive PTO probes are sent. I am not sure if it works well across the board - it would not work when we suddenly observe packets being dropped on a stable condition.

I would rather considering fixing that issue rather than discussing about if we should make this code look better.

Reply:
> I am not sure if it works well across the board - it would not work when we suddenly observe packets being dropped on a stable condition.

Let me retract this. The code works fine under such situation, because `cc_bytes_in_flight` would remain as non-zero while PTO probes are sent consecutively; we never hit this condition.

This condtion (`loss->sentmap.num_packets < 32`) purely exists for improving the chance of making forward progress under heavy ACK loss. 32 has been chosen as it is enough to cover multiple rounds of PTO.

Reply:
Okay I think I'm convinced. I would add a comment to summarize above discussion for future reference.

Reply:
Added some more explanation in 08782e5. PTAL.

Reply:
👍 looks good, thanks!